stream.go

```diff
@@ -112,7 +120,13 @@ func (stream *Stream) connect() (r io.ReadCloser, err error) {
 
 func (stream *Stream) stream(r io.ReadCloser) {
 	defer r.Close()
+	defer stream.wg.Done()
+
+	var deadline time.Time
+	lastSuccess := time.Now()
 	dec := NewDecoder(r)
+	context := stream.req.Context()
+	timeout, hasTimeout := context.Value("timeout").(time.Duration)
```

Comment:
The more idiomatic way to indicate a timeout (or cancellation or exceeded deadline) these days should be the [Done channel on the Context](https://golang.org/pkg/context/#Context). Is there any particular reason we're not using that instead?